src/index.js

```diff
@@ -98,10 +172,45 @@ function listTopMovies(input) {
         .addClass('.error')
         .text(messages[Error.NO_RESULTS])
         .fadeIn('slow');
+      delayFlashHide();
     });
 }
 
-function fetchMovies(year) {
-  let url = urlJoin(proxyUrl, baseUrl, '/year/', year);
-  return fetchDocument(url).then(doc => parseMovies(doc));
+/* infinite scroll */
+
+$window.scroll(e => {
+  if ($(e.target).scrollTop()) {
+    $arrowToTop.fadeIn();
+  } else {
+    $arrowToTop.fadeOut();
+  }
```

Comment:
It's not possible to apply there. I just realized that as long as the if statement is true (and that is all the time except in one case when window.scrollTop === 0 ) arrow gets rerendered, that's a lot of events fired to waste, I fixed it now with the render flag. Now it gets called only once.